sqlmesh/core/snapshot/definition.py

```diff
@@ -213,22 +219,36 @@ def catalog_for_environment(
     ) -> t.Optional[str]:
         return environment_naming_info.catalog_name_override or self.catalog
 
-    def schema_for_environment(self, environment_naming_info: EnvironmentNamingInfo) -> str:
+    def schema_for_environment(
+        self, environment_naming_info: EnvironmentNamingInfo, dialect: DialectType = None
+    ) -> str:
         schema = self.schema_name or c.DEFAULT_SCHEMA
         if (
             environment_naming_info.name.lower() != c.PROD
             and environment_naming_info.suffix_target.is_schema
         ):
-            schema = f"{schema}__{environment_naming_info.name}"
+            env_name = environment_naming_info.name
+            if environment_naming_info.normalize_name:
+                env_name = normalize_identifiers(env_name, dialect=dialect).name
+
+            schema = f"{schema}__{env_name}"
```

Comment:
how does this work? do we not normalize schema? doesn't that mean it could be

sqlmesh_DEV ?

Reply:
or default_DEV

Reply:
Ah thanks, I think this is an oversight. I'll do another pass at this soon and make sure we normalize everything.

Reply:
I think we should only normalize the `default` schema, right? If a user's set their own schema, then that should have been normalized prior to flowing into this method. I verified this in my tests because `sqlmesh plan foo` ended up creating `FOO__DEV`.

Reply:
Addressed